src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -199,6 +203,42 @@ public String getName() {
     return "remote";
   }
 
+  class ExecutingStatusReporter implements ExecuteOperationUpdateReceiver {
+    private boolean reportedExecuting = false;
+    private SpawnExecutionContext context;
+
+    ExecutingStatusReporter(SpawnExecutionContext context) {
+      this.context = context;
+    }
+
+    @Override
+    public void onNextOperation(Operation o) throws IOException {
+      if (!reportedExecuting) {
+        if (o.hasMetadata()) {
```

Comment:
I'd recommend `o.getMetadata().is(ExecuteOperationMetadata.class)` instead or Bazel will crash if the server sends something that isn't this proto.